src/ui/shared/truncateAddress.test.ts

```diff
@@ -0,0 +1,28 @@
+import { truncateAddress } from './truncateAddress';
+import { ellipsis } from './typography';
+
+describe('truncateAddress', () => {
+  test('with 0x prefix', () => {
+    const address = '0x7358B726830A2E222f9b139E90483A37142bcBE5';
+    const expected = `0x7358B7${ellipsis}2bcBE5`;
+    expect(truncateAddress(address)).toBe(expected);
+  });
+
+  test('without 0x prefix', () => {
+    const address = '7358B726830A2E222f9b139E90483A37142bcBE5';
+    const expected = `7358B7${ellipsis}2bcBE5`;
+    expect(truncateAddress(address)).toBe(expected);
+  });
+
+  test('custom padding', () => {
+    const address = '0x7358B726830A2E222f9b139E90483A37142bcBE5';
+    const expected = `0x7358${ellipsis}cBE5`;
+    expect(truncateAddress(address, 4)).toBe(expected);
+  });
```

Comment:
Let's test at least two different padding values?

Reply:
added 2 more tests with 4 and 8 padding